src/librustdoc/clean/mod.rs

```diff
@@ -2682,11 +2668,12 @@ fn name_from_pat(p: &hir::Pat) -> String {
     match p.node {
         PatKind::Wild => "_".to_string(),
         PatKind::Binding(_, ref p, _) => p.node.to_string(),
-        PatKind::TupleStruct(ref p, ..) | PatKind::Path(None, ref p) => path_to_string(p),
+        PatKind::TupleStruct(ref p, ..) | PatKind::Path(None, ref p) => p.to_string(),
```

Comment:
`hir::print::State::print_path` begins with `self.maybe_print_comment(path.span.lo)`, which might actually print a comment in this case:
```rust
fn f(arg1: (), // Some comment
     path::to::TupleStruct(x): path::to::TupleStruct) {
     //^ The `name_from_pat` of this tuple struct pattern might include "// Some comment"
}
```